src/android/CameraPreviewFragment.java

```diff
@@ -1,109 +1,208 @@
-        camera.addCameraListener(new CameraListener() {
-            @Override
-            public void onCameraOpened(CameraOptions options) {
-                if (startCameraCallback != null)
-                    startCameraCallback.onCameraStarted();
-            }
-            @Override
-            public void onPictureTaken(PictureResult result) {
-                UUID uuid = UUID.randomUUID();
-                File file = new File(new ContextWrapper(getActivity().getBaseContext()).getFilesDir(), uuid.toString() + ".jpg");
-                result.toFile(file, (File mfile)->{
-                    if (mfile == null){
-                        capturePictureCallback.onCompleted(new Exception("unable to save image"), null);
-                    }
-                    capturePictureCallback.onCompleted(null, mfile.getName());
-                });
 
-            }
-        });
-        camera.setLayoutParams(new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT));
-        containerView.addView(camera);
+        viewFinder = new PreviewView(getActivity());
+        viewFinder.setLayoutParams(new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT));
+        containerView.addView(viewFinder);
+        startCamera();
+
         return containerView;
     }
 
     @Override
-    public void onResume() {
-        super.onResume();
-        camera.open();
+    public void onStart() {
+        super.onStart();
+
+        if (lifecycleRegistry.getCurrentState() == Lifecycle.State.CREATED) {
+            lifecycleRegistry.setCurrentState(Lifecycle.State.STARTED);
+        }
+    }
+
+    public void startCamera() {
+        ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(getActivity());
+        ProcessCameraProvider cameraProvider = null;
+
+        try {
+            cameraProvider = cameraProviderFuture.get();
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            Log.e(TAG, "startCamera: " + e.getMessage());
+        }
+
+        CameraSelector cameraSelector = new CameraSelector.Builder()
+                .requireLensFacing(direction)
+                .build();
+        preview = new Preview.Builder().build();
+        imageCapture = new ImageCapture.Builder().build();
+
+        cameraProvider.unbindAll();
+        camera = cameraProvider.bindToLifecycle(
+                this::getLifecycle,
+                cameraSelector,
+                preview,
+                imageCapture
+        );
+
+        preview.setSurfaceProvider(viewFinder.getSurfaceProvider());
+
+        if (startCameraCallback != null) {
+            startCameraCallback.onCameraStarted();
+        }
+    }
+
+    public void takePicture(boolean useFlash, CameraCallback takePictureCallback) {
+        this.capturePictureCallback = takePictureCallback;
```

Comment:
Ok here we should try to remove the this.capturePictureCallback. So we do not use any global variable to store the callback (it can create side effects)